From e218da8cd707644baf120a8874eca1ae192d8cc8 Mon Sep 17 00:00:00 2001
From: Pavel Snajdr <snajpa@snajpa.net>
Date: Thu, 5 Sep 2019 23:15:14 +0200
Subject: [PATCH 10/13] [vpsAdminOS] Attempt to fix our arc_prune storms

These are especially bad when zfs_arc_meta_strategy=META_ONLY
---
 module/zfs/arc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/module/zfs/arc.c b/module/zfs/arc.c
index 9c5ee9829..19ea67aa9 100644
--- a/module/zfs/arc.c
+++ b/module/zfs/arc.c
@@ -3919,7 +3919,8 @@ arc_evict_state(arc_state_t *state, uint64_t spa, int64_t bytes,
 		 * Request that 10% of the LRUs be scanned by the superblock
 		 * shrinker.
 		 */
-		if (type == ARC_BUFC_DATA && aggsum_compare(&astat_dnode_size,
+		if (type == ARC_BUFC_METADATA &&
+		    aggsum_compare(&astat_dnode_size,
 		    arc_dnode_size_limit) > 0) {
 			arc_prune_async((aggsum_upper_bound(&astat_dnode_size) -
 			    arc_dnode_size_limit) / sizeof (dnode_t) /
-- 
2.20.1

